examples/delete-user-identity.ts

```diff
@@ -0,0 +1,126 @@
+import type { Builder, Command, Describe } from 'landlubber'
+
+import type { AcsCredentialsListResponse } from '@seamapi/http/connect'
+
+import type { Handler } from './index.js'
+
+interface Options {
+  userIdentityId: string
+}
+
+export const command: Command = 'delete-user-identity userIdentityId'
+
+export const describe: Describe = 'Delete a user identity'
+
+export const builder: Builder = {
+  userIdentityId: {
+    type: 'string',
+    describe: 'User identity id to delete',
+  },
+}
+
+export const handler: Handler<Options> = async ({
+  userIdentityId,
+  seam,
+  logger,
+}) => {
+  const waitForEvent =
+    (eventType: string, idKey: string) =>
+    async (resource: any): Promise<void> => {
+      // Application architecture specific logic that waits for an event matching
+      // event.event_type === eventType && event[idKey] === resource[idKey]
+      // In this example, the event may arrive before this function is called,
+      // so any implementation must handle that case.
+      //
+      // More generally, an application just needs to wait for the resource to be deleted before continuing,
+      // or the next operation may error.
+      // If, on error, this entire deletion function is called again repeatedly,
+      // deletion will either eventually succeed, or be stuck due to some resource that fails to delete.
+      logger.info({ eventType, idKey, resource }, 'Got event')
+    }
+
+  const clientSessions = await seam.clientSessions.list({
+    // param not implemented
+    user_identity_id: userIdentityId,
+  })
+
+  for (const clientSession of clientSessions) {
+    await seam.clientSessions.delete({
+      client_session_id: clientSession.client_session_id,
+    })
+  }
+
+  const enrollmentAutomations =
+    await seam.userIdentities.enrollmentAutomations.list({
+      user_identity_id: userIdentityId,
+    })
+
+  for (const enrollmentAutomation of enrollmentAutomations) {
+    // endpoint not implemented
+    await seam.userIdentities.enrollmentAutomations.delete({
+      enrollment_automation_id: enrollmentAutomation.enrollment_automation_id,
+    })
+  }
+
+  await Promise.all(
+    enrollmentAutomations.map(
+      // event not implemented
+      waitForEvent('enrollment_automation.deleted', 'enrollment_automation_id'),
```

Comment:
Hmm i wish we could poll thr deleted resource to avoid forcing event-based, but i think we're aligned on that

Reply:
Agreed, however as we discussed the API behavior around 404 on resources still being deleted needs to be made consistent first.

You gave me another idea though for this example: implement this function via polling `/events/list`

Reply:
100% full alignment, resolving